src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -516,7 +528,7 @@ private async Task<Compilation> GetOrBuildDeclarationCompilationAsync(SolutionSt
                     compilation = compilation.AddSyntaxTrees(trees);
                     trees.Free();
 
-                    WriteState(new FullDeclarationState(compilation), solution);
+                    WriteState(new FullDeclarationState(compilation, new TrackedGeneratorDriver(generatorDriver: null)), solutionServices);
```

Comment:
Nit: rest of PR uses `new TrackedGenerator(null)`. Not opposed to the name but seems like it should be consistent. 

As an aside the pattern seems odd. Feels like it would be better as a named factory of sorts say `TrackedGeneratorDriver.CreateNeedsFullGeneration`

Reply:
TrackedGeneratorDriver is a workaround for a API problem right now with GeneratorDriver. GeneratorDriver has a way to reuse a previous driver with a known set of edits, which you can then apply with TryApplyEdits. But if there's some change that happens to a compilation that doesn't have a corresponding edit, it's just up to the user of the driver to remember to not call TryApplyEdits. I've asked @chsienki to instead offer some API to "poison" a generator driver which would mean:

1. Any stored edits are tossed, because there's no point in holding onto them anymore. (It's otherwise potentially just leaking memory.)
2. Any future calls to TryApplyEdits just fails, so a full generation has to be ran instead. Since the driver requires that as a fallback no matter what (the host has to be ready for TryApplyEdits to fail at any time, so this isn't extra work for us.)

At this point the struct is just going away.